src/edit/main.js

```diff
@@ -186,6 +186,17 @@ class ProseMirror {
     this.options.keymaps.forEach(map => this.addKeymap(map, -100))
 
     this.options.plugins.forEach(plugin => plugin.attach(this))
+
+    // get the root of prosemirror since it could be in a shadow dom
+    if (opts.place) {
```

Comment:
Is it possible for "place" to be added later via a setter? If it can, I presume we want this logic to be executed on that code path too.

Perhaps having a setter for `place` that calculates `root` at the time of assignment would solve this.